vignettes/loo2-moment-matching.Rmd

```diff
@@ -0,0 +1,309 @@
+to be able to evaluate the predictive performance of the model accurately.
+
+## Moment matching correction for importance sampling
+
+To improve the accuracy of the `loo()` result above, we could perform
+leave-one-out cross-validation by explicitly leaving out single observations
+and refitting the model using MCMC repeatedly. However, the Pareto $k$ diagnostics
+indicate that there are 19 observations which are problematic. This would
+require 19 model refits which may require a lot of computation time.
+
+Instead of refitting with MCMC, we can perform a faster moment matching correction to
+the importance sampling for the problematic observations. 
+This can be done with the `loo_moment_match()` function in the
+__loo__ package, which takes our existing `loo` object as input
+and modifies it. The moment matching requires some evaluations of the model
+posterior density. For models fitted with __rstan__, this can be conveniently
+done by using the existing `stanfit` object.
+
+First, we show how the moment matching can be used for a model fitted
+using __rstan__. It only requires setting the argument `moment_match` to `TRUE`
+in the `loo()` function.
+Optionally, you can also set the argument `k_threshold` which determines
+the Pareto $k$ threshold, above which moment matching is used.
+By default, it operates
+on all observations whose Pareto $k$ value is larger than 0.7.
+(NOTE: Using moment matching directly using the `loo()` function is currently
+available in the development version of __rstan__ in
+[ ](https://github.com/topipa/rstan), but will
+be available in the CRAN release eventually).
+
+```{r, message=FALSE}
+loo2 <- loo(fit, moment_match = TRUE)
+loo2
+```
+
+After the moment matching, all observations have the diagnostic
+Pareto $k$ less than 0.7, meaning that the estimates are now reliable.
+The total `elpd_loo` estimate also changed from `-5457.8` to `-5478.5`, showing
+that before moment matching, `loo()` overestimated the predictive
+performance of the model.
+
+The updated Pareto $k$ values stored in
+`loo2$diagnostics$pareto_k` are considered algorithmic diagnostic values that
+indicate the sampling accuracy. The original Pareto $k$ values are
+stored in `loo2$pointwise[,"influence_pareto_k"]` and these are not modified
+by the moment matching. These can be considered as diagnostics
+for how big influence each observation has on the posterior distribution.
+In addition to the Pareto $k$ diagnostics, moment matching also updates
+the effective sample size estimates. 
+
+# Using `loo_moment_match()` directly
+
+The moment matching can also be performed by explicitly calling the function
+`loo_moment_match()`. This enables its use also for models that
+are not using __rstan__.
+To use `loo_moment_match()`, the user must give the model object `x`, the `loo`
+object, and 5 helper functions as arguments to
+`loo_moment_match()`.
+These helper functions, in turn, take `x` as an argument.
```

Comment:
Since you explain the arguments of each function below, stating here that these helper functions take `x` as an argument is redundant and might be misleading since we have more arguments.